src/app/services/tasksService/tasks.service.ts

```diff
@@ -117,6 +117,38 @@ export class TasksService {
         } else {
           observer.error(new HttpResponse({ status: 404, statusText: 'No item with such ID found!' }));
         }
+      }, 2000);
+    });
+  }
+  deleteActionItem(itemId: string): void {
+    setTimeout(() => {
+      this.dataTable.filter(item => {
+        if (item.id === itemId) {
+          this.dataTable.splice(this.dataTable.indexOf(item), 1);
+        }
+      }, 1000);
+    });
+  }
+  editActionItem(
```

Comment:
**CONCERN:** In my opinion, we should pass here the `EditActionItemModel` similarly as in the add action item. Then you could benefit and simplify the code with spread operator. It's also easier to extend the class instead of adding additional parameter (and imho more readable).